SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc

```diff
@@ -1971,3 +2001,452 @@ void SiPixelDigitizerAlgorithm::module_killing_DB(uint32_t detID) {
+    std::cout << "Cluster after reweighting: " << std::endl;  
+    printCluster(pixrewgt);
+  }
+  
+  for(int row = 0; row < TXSIZE; ++row) { 
+    for(int col = 0; col < TYSIZE; ++col) {
+      float charge = 0;
+      charge = pixrewgt[row][col];
+      chargeAfter += charge;
+      if( (hitPixel.first + row - THX) >= 0 && (hitPixel.first + row - THX) < topol->nrows() && (hitPixel.second + col - THY) >= 0 && (hitPixel.second + col - THY) < topol->ncolumns() && charge > 0)
+	theSignal[PixelDigi::pixelToChannel(hitPixel.first + row - THX, hitPixel.second + col - THY)] += (makeDigiSimLinks_ ? Amplitude(charge , &hit, hitIndex, tofBin, charge) : Amplitude( charge, charge) )  ;
+    }
+  }
+  
+  if(PrintClusters){
+    std::cout << std::endl;
+    std::cout << "Charges (before->after): " << chargeBefore << " -> " << chargeAfter << std::endl;
+    std::cout << "Charge loss: " << (1 - chargeAfter/chargeBefore)*100 << " %" << std::endl << std::endl;
+  }
+
+  return true;
+
+}
+
+// *******************************************************************************************************
+//! Reweight CMSSW clusters to look like clusters corresponding to Pixelav Templates.     
+//! \param       id_in - (input) identifier of the template corresponding to the input events                                 
+//! \param    id_rewgt - (input) identifier of the template corresponding to the output events                                 
+//! \param     cluster - (input/output) boost multi_array container of 7x21 array of pixel signals, 
+//!                       origin of local coords (0,0) at center of pixel cluster[3][10].                      
+//! returns 0 if everything is OK, 1 if angles are outside template coverage (cluster is probably still
+//! usable, > 1 if something is wrong (no reweight done).
+// *******************************************************************************************************
+int SiPixelDigitizerAlgorithm::PixelTempRewgt2D(int id_in, int id_rewgt, array_2d& cluster)
+{
+  // Local variables 
+  int i, j, k, l, kclose;
+  int nclusx, nclusy, success;
+  float xsize, ysize, q50i, q100i, q50r, q10r, q100r, xhit2D, yhit2D, qclust, dist2, dmin2;
+  float xy_in[BXM2][BYM2], xy_rewgt[BXM2][BYM2], xy_clust[TXSIZE][TYSIZE];
+  int denx_clust[TXSIZE][TYSIZE], deny_clust[TXSIZE][TYSIZE];
+  int goodWeightsUsed, nearbyWeightsUsed, noWeightsUsed;
+  float cotalpha, cotbeta; 
+  // success = 0 is returned if everthing is OK
+  success = 0;
+    
+  // Copy the array to remember original charges
+  array_2d clust(cluster);
+
+  // Take the pixel dimensions from the 2D template
+  xsize = templ2D.xsize();
+  ysize = templ2D.ysize();
+  
+  // Calculate the track angles
+
+  if (fabs(track[5]) > 0.){
+    cotalpha = track[3]/track[5]; //if track[5] (direction in z) is 0 the hit is not processed by re-weighting
+    cotbeta = track[4]/track[5];
+  } else {
+    std::cout << "Reweighting angle is not good!" << std::endl;
```

Comment:
This is run time method (?) where std::cout should not be used, use instead edm:LogWarning